src/constructs/aws/ServerSideWebsite.ts

```diff
@@ -372,7 +372,7 @@ export class ServerSideWebsite extends AwsConstruct {
     private createRequestFunction(): cloudfront.Function {
         let additionalCode = "";
 
-        if (this.configuration.redirectToMainDomain === true) {
+        if (this.configuration.redirectToMainDomain === true && this?.domains && this.domains.length > 1) {
```

Comment:
The build still fails, could it be that `this?` is invalid because `this` cannot be null?